src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1027,17 +1027,17 @@ static BoundExpression bindCollectionConstructorConstruction(
                     {
                         var binder = new ParamsCollectionTypeInProgressBinder(namedType, @this._binder, withElement != null, constructor);
                         collectionCreation = binder.BindClassCreationExpression(syntax, namedType.Name, syntax, namedType, analyzedArguments, @this._diagnostics);
-                        collectionCreation.WasCompilerGenerated = true;
                     }
                     else if (@this._targetType is TypeParameterSymbol typeParameter)
                     {
                         collectionCreation = @this._binder.BindTypeParameterCreationExpression(syntax, typeParameter, analyzedArguments, initializerOpt: null, typeSyntax: syntax, wasTargetTyped: false, @this._diagnostics);
-                        collectionCreation.WasCompilerGenerated = true;
                     }
                     else
                     {
                         throw ExceptionUtilities.UnexpectedValue(@this._targetType);
                     }
+
+                    collectionCreation.WasCompilerGenerated = withElement is null;
```

Comment:
intentional change.  so that GetSymbolInfo on a with-element can find this node.